src/io/openshift/plugins/analytics.groovy

```diff
@@ -5,6 +5,37 @@ import io.openshift.Events
 def register() {
     Events.on("build.pass") {
         e, a -> echo "invoking bayesian analytics $e $a"
+        def image = config.runtime() ?: 'node'
+        spawn (image: image, version: config.version()) {
+          echo "Pod spawned."
+          def url = Utils.shWithOutput(this, "git config remote.origin.url")
+          echo "URL:$url"
+          image = config.runtime() ?: 'maven'
+          echo "Image:$image"
+          
+          if(image == "node") {
+              sh '''
+                  echo "running additional npm commands"
+                  npm install --prod
+                  npm list --prod --json > direct-dependencies.txt
+              '''
+          }
+          else {
+            echo "Image type not handled: $image"
+          }
+          try {
+              retry(3) {
+                  def response = bayesianAnalysis url: 'https://bayesian-link', gitUrl: url
+                  if (response.success) {
+                    Utils.addAnnotationToBuild(this, 'fabric8.io/bayesian.analysisUrl', response.getAnalysisUrl())
+                  } else {
+                    error "Bayesian analysis failed ${response}"
```

Comment:
this will stop the entire pipeline and I don't think failing analytics should stop the pipeline

Reply:
+1

Reply:
A echo with a message is  suffice here, we can fail the complete pipeline just because it failed for analytics